tracer/src/Datadog.Trace/AppSec/Coordinator/SecurityCoordinator.Framework.cs

```diff
@@ -386,6 +385,31 @@ private bool WriteAndEndResponse(BlockingAction blockingAction)
         return dict;
     }
 
+    public Dictionary<string, string[]> GetResponseHeadersForWaf()
+    {
+        var response = _context.Response;
+        var headersDic = new Dictionary<string, string[]>(response.Headers.Keys.Count);
+        var headerKeys = response.Headers.Keys;
+        foreach (string originalKey in headerKeys)
+        {
+            var keyForDictionary = originalKey ?? string.Empty;
+            if (!keyForDictionary.Equals("cookie", System.StringComparison.OrdinalIgnoreCase))
+            {
+                keyForDictionary = keyForDictionary.ToLowerInvariant();
```

Comment:
Little Nit
```suggestion
            var keyForDictionary = originalKey?.ToLowerInvariant() ?? string.Empty;
            if (keyForDictionary != "cookie")
            {
```

Reply:
Good idea, thanks! Committed in 0ce2bcc51